### PR TITLE
DATA-859 add motion benchmarks to CI

### PR DESCRIPTION
**Pull request by raybjork**
This PR adds some new workflows to github actions that integrate with recent work done in the `motion-testing` library in the name of providing continuous testing for the `motionplan` package we have been developing in RDK.  In summary, this workflow will fire on a pull request to main that changes files in one of the following packages, which are downstream of this pacakge:
- `motionplan`
- `referenceframe`
- `spatialmath`

A separate action will fire upon completion that comments on the relevant PR with a readout of the test results.  

For a demonstration of these actions, please see the [associated PR](https://github.com/viamrobotics/rdk/pull/1779) which demonstrates that changing the number of solutions to seed  from 50->1 is in fact worse for motion planning.

Also contingent on this PR being merged is [another associated PR in `motion-testing`](https://github.com/viamrobotics/motion-testing/pull/2).  

When this PR has approval, I will go back and merge in the PR to motion-testing, then change the associated branch references to "main" (which I will call out in comments) before finally merging into RDK.

### CODE REVIEW

File: referenceframe/frame_system.go
Comment (biotinker) [inline referenceframe/frame_system.go:343]: Why are we deleting comments out of framesystem?

File: .github/workflows/pr-labeler.yml
Comment (biotinker) [inline .github/workflows/pr-labeler.yml:5]: This would need to be undone, right?

File: .github/workflows/motion-pull-request-trusted.yml
Comment (biotinker) [inline .github/workflows/motion-pull-request-trusted.yml:9]: This will need to be updated to `main` I think

**Review by biotinker — COMMENTED** [review]
Not super familiar with github workflows but left a few comments

**Comment by raybjork** [comment]
@biotinker I left comments for myself of what will change before final merge

File: .github/workflows/motion-benchmarks-comment.yml
Comment (biotinker) [inline .github/workflows/motion-benchmarks-comment.yml:34]: file should have newline

**Review by biotinker — APPROVED** [review]
This LGTM modulo the comments being updated

File: .github/workflows/motion-benchmarks.yml
Comment (edaniels) [inline .github/workflows/motion-benchmarks.yml:8]: remove

**Review by edaniels — APPROVED** [review]
LGTM once lints are fixed and other things we talked about offline are addressed!

**Pull request merged**
